[FIX] Refine antsBrainExtraction if recon-all is run

oesteban opened the pull request:
Use the `aseg.mgz` output of FreeSurfer to derive a refined mask to replace the (generally less conservative) mask calculated with `antsBrainExtraction.sh`.

Closes #431

oesteban commented:
Hold on with the review, I'm adding a little improvement.

oesteban commented on fmriprep/interfaces/freesurfer.py:
```diff
@@ -246,3 +285,70 @@ def detect_inputs(t1w_list, t2w_list=None, hires_enabled=True):
+    indices = np.argwhere(newrefmask > 0)
+    for pixel in indices:
+        # When ATROPOS identified the pixel as GM, set and carry on
+        if ants_segs[tuple(pixel)] == 2:
```
These lines incorporate Arno's concepts for relabeling pixels excluded by FreeSurfer. However, my impression is that this adds a lot of pixels wrongly identified as GM by ATROPOS during `antsBrainExtraction.sh` (the segmentation comes from there).

Let me know if we want to keep it or discard it.

chrisfilo replied:
It would be useful to see an example of this in action (in other words which voxels are wrongly relabeled).

oesteban replied:
Okay, I'll get some examples.

I have to say that Arno's segmentations are probably a lot better, since he gets them from `antsCorticalThickness.sh`. The idea is probably great, but currently we can't apply it properly.

oesteban replied:
https://github.com/oesteban/notebooks/blob/f5cf852d212867b32e7bf38bda2424e7cc132a35/FMRIPREP-issue431.ipynb - discussion moved to the main thread.

chrisfilo commented:
Let me know when this is ready for review.

oesteban commented:
Here, two examples.

**Legend**: yellow is the original ANTs mask; red is with Arno's relabeling; green is the refined mask without Arno's relabeling.

### ds000114/sub-01 (the first example of #431).

![ds114](https://user-images.githubusercontent.com/598470/34398217-f497a018-eb31-11e7-9373-66204c426b50.png)


### ds000017/sub-2 (second example)
![ds017-1](https://user-images.githubusercontent.com/598470/34398377-55f57a5a-eb33-11e7-961e-2fdc99f2c0bf.png)
![ds017-2](https://user-images.githubusercontent.com/598470/34398378-560c11f2-eb33-11e7-9a97-a5d0f875f233.png)

It seems like, despite my first impression, the version with Arno's trick may work a bit better for these two cases. In general, both solutions seem better than the original mask (yellow contour).

chrisfilo commented:
Both solutions look comparable (each makes mistakes at different locations). The improvement over ants is great.

One more thing I would check is how this new - tighter - mask works with normalization. If the template mask is more liberal it could be problematic.

oesteban commented:
Sweet. Should we just run 10 subjects to check the impact on spatial normalization? WDYT?

chrisfilo commented:
Sounds like a good plan.

Best,
Chris

On Dec 29, 2017 10:48 PM, "Oscar Esteban" <notifications@github.com> wrote:

> Sweet. Should we just run 10 subjects to check the impact on spatial
> normalization? WDYT?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/poldracklab/fmriprep/pull/912#issuecomment-354531032>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOkp1j4YJDg7JXPfLOayerRA-aOh2G-ks5tFdyrgaJpZM4RNJ8I>
> .
>

oesteban commented:
Done, please wait for Circle to generate documentation. The diff is not particularly helpful since I introduced some line breaks for readability. Some changes are unrelated to this PR, but I added them because I judged it preferable (even if it means holding off merging for a little while):

  - [x] Added reference to Nilearn.
  - [x] Better described what are tools and what are packages (using the `<code>tool</code>` tags for tools)
  - [x] Brain tissue segmentation moved to the anatomical processing
  - [x] Rewrote the CompCor passage.
  - [x] Minor nits here and there (e.g. removing the `[Internet]` marker added after the journal names on references).

Also, updated the CHANGES file with all PR done so far since 1.0.0, with the idea of making it easier to brew a new release quickly and soon.

oesteban commented:
I'm not seeing any major problem with the extra 10 subjects and their masks. I haven't identified any difference in spatial normalization either. Merging...
